### ISSUE TITLE
IO/Task false positives

### ISSUE THREAD

**Issue by jleider** [buildo/eslint-plugin-fp-ts#62]
#55 is amazing, I cant tell you how many times I have forgotten to call an IO/Task, this linter rule already found 1 bug and 1 use of a `TaskEither` that was unnecessary! 

However, I also ran into a couple false positives:

I am getting this linter warning: ```fp-ts/no-discarded-pure-expression: 'IORef' is pure, so this expression does nothing in statement position. Did you forget to return it or run it?``` within the constructor of a class where its being assigned to a class property. I imagine all 4 warnings are all related.

Example (not the actual code):
```typescript
export interface ErrorResponseInterface<E = string> {
  readonly field: string;
  readonly error: E;
  readonly extra?: string;
}

export class Foo {
  readonly getErrors: IO<ErrorResponseInterface[]>;
  readonly visibleErrors: IO<ErrorResponseInterface[]>;
  readonly isValid: IO<boolean>;
  private readonly errors: IORef<ErrorResponseInterface[]>;

  // Warning on all 4 lines within this constructor
  constructor(settings: BasilSettings) {
    this.errors = new IORef([]);
    this.getErrors = this.errors.read;
    this.visibleErrors = pipe(this.getErrors, mapIO(invoke1("filter")(flow(this.inputOrLabelForError, isSome))));
    this.isValid = pipe(this.visibleErrors, mapIO(flow(prop("length"), eq(0))));
  }

  ...
}
```

**Comment by gabro** [comment]
Hi @jleider, thanks for reporting!

> this linter rule already found 1 bug and 1 use of a TaskEither that was unnecessary!

🤩 that's great to hear!

> However, I also ran into a couple false positives

😢 that's not so great, but fortunately the fix is really simple. I've overlooked that the type of assignement expressions resolves to the assigned type, but in general we shouldn't care about assignements for this rule because if you're assigning something you're not accidentally discarding it (unless the assignee is unused, but that's already covered by other rules and TS itself).

I'll send a fix right away, thanks again for the report and the reproduction!

**Comment by gabro** [comment]
Release as v0.2.1 :shipit:

**Comment by jleider** [comment]
Yep, this fixed my false positives, thanks for the quick turnaround!
